### ISSUE TITLE
new logback versions do not log to file 

### ISSUE THREAD

**Issue by chas678** [dropwizard/dropwizard#1478]
Dropwizard 0.9.2 uses Logback 1.1.3.  Attempting to use newer versions of Logback 1.1.5 + 1.1.6 mean that logs are longer written to file when configured to do so in the YAML which violates the principle of least astonishment.

## Steps to reproduce

1. change pom to use logback version 1.1.5 or 1.1.6
1. Run a service locally with a file appender configured in the YAML - for example: 

```yaml
logging:
  level: INFO
  loggers:
    "com.example": DEBUG
  appenders:
    - type: console
      target: stdout
      timeZone: UTC
      logFormat: "%-5p [%d{ISO8601,UTC}] %7X{userUUID} %c: %m%n%rEx"
    - type: file
      threshold: ALL
      currentLogFilename: /tmp/example-service/server.log
      archivedLogFilenamePattern: /tmp/app/example-service/server-%d.log.gz
      archivedFileCount: 20
      timeZone: UTC
      logFormat: "%-5p [%d{ISO8601,UTC}] %7X{userUUID} %c: %m%n%rEx"
```
## Expected results
* Log goes to console and also to file at ` /tmp/example-service/server.log`

## Actual results
* log goes to console, but no file is created in ` /tmp/example-service/server.log`

## Suspicion/Guess as to cause
Logback has changed the way MDC is inherited by threads .

## Refs
* Logback release notes : http://logback.qos.ch/news.html

**Comment by xiaodong-xie** [comment]
I don't think it's related to MDC.

Here is my analysis:
1. It's because `RollingFileAppender` is more strict from 1.1.5, as it will check whether the `triggeringPolicy` is started or not.
2. In class `FileAppenderFactory`, we first `stop` then `start` the appender, so the `triggeringPolicy` is stopped. 
3. I managed to pass all test cases in `dropwizard-logging` module by just removing line 202 `appender.stop();`.

So my question is, is `appender.stop();` really necessary? 

Here is my PR: https://github.com/dropwizard/dropwizard/pull/1481

**Comment by ghenkes** [comment]
I looked into this some last week as well and came to the same conclusion as @xiaodong-xie, that `appender.stop()` should be removed.

When `stop` is called we eventually get to the `stop` method in OutputStreamAppender, which has Javadoc that states: [Stopped appenders cannot be reused.](https://github.com/qos-ch/logback/blob/v_1.1.6/logback-core/src/main/java/ch/qos/logback/core/OutputStreamAppender.java#L108) So it looks like calling `stop` may have worked ok prior to Logback 1.1.5, but I don't think it was intended to.

In addition to the change in @xiaodong-xie's pull request there also needs to be a change to the [DefaultLoggingFactoryTest::testConfigure](https://github.com/dropwizard/dropwizard/blob/master/dropwizard-logging/src/test/java/io/dropwizard/logging/DefaultLoggingFactoryTest.java#L91) class and corresponding [logging_advanced.yml](https://github.com/dropwizard/dropwizard/blob/master/dropwizard-logging/src/test/resources/yaml/logging_advanced.yml#L22) test resource, because the `com.example.newApp` and `com.example.notAdditive` appenders are writing to the same log file which is [not allowed by Logback](https://github.com/qos-ch/logback/blob/v_1.1.6/logback-core/src/main/java/ch/qos/logback/core/rolling/RollingFileAppender.java#L62).

**Comment by ghenkes** [comment]
Also, I'd caution that it looks like Logback 1.1.5 and 1.1.6 has introduced a few [new bugs](http://jira.qos.ch/browse/LOGBACK-1154?jql=project%20%3D%20LOGBACK%20AND%20issuetype%20%3D%20Bug%20AND%20resolution%20%3D%20Unresolved%20AND%20affectedVersion%20in%20(1.1.5%2C%201.1.6)%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC) so I'm not sure at what point the Dropwizard Committers want to upgrade the Logback version, or if they'd rather wait for these to be fixed. I've tried out 1.1.6 with the corresponding changes to dropwizard that I've mentioned above and didn't see any issues, but that may just be for my use case.

**Comment by xiaodong-xie** [comment]
Thanks, @ghenkes. :)

Noticed what you described in your comment, when upgrading Logback to 1.1.6. So I just added another commit to fix the test case. But you're right about those new bugs introduced in Logback 1.1.5 and 1.1.6 of course.

**Comment by arteam** [comment]
Thanks you lot for a deep analysis of the issue! Dropwizard master branch has been updated to Logback 1.1.6. The upgrade will be available in Dropwizard 1.0.0-rc2.

**Comment by DNewfield** [comment]
This issue was initially reported against version 0.9.2, but was only fixed in 1.0.0...
..would you accept a patch (basically the same one previously offered) against branch release/0.9.x to address this there, too?  The downside is that would necessitate releasing 0.9.4 to get the fix out...
